go/consensus/tendermint/apps/staking/transactions.go

```diff
@@ -315,26 +314,14 @@ func (app *stakingApplication) reclaimEscrow(ctx *api.Context, state *stakingSta
 		return nil, staking.ErrForbidden
 	}
 
-	to, err := state.Account(ctx, toAddr)
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch account: %w", err)
+	if !toAddr.Equal(reclaim.Account) && params.DisableDelegation {
+		return nil, staking.ErrForbidden
 	}
 
 	// Fetch escrow account.
-	//
-	// NOTE: Could be the same account, so make sure to not have two duplicate
-	//       copies of it and overwrite it later.
-	var from *staking.Account
-	if toAddr.Equal(reclaim.Account) {
-		from = to
```

Comment:
This was deliberate if somewhat wonky, but `to` is never mutated, so the removal appears to be ok.

Reply:
right, to prevent creating multiple in-memory representations of the same record that can get out of sync. in the changed version, we no longer attempt to load the "to"/delegator account